tests/test_resources.py

```diff
@@ -213,68 +324,143 @@ def test_user_albums(self, client):
         """
         Test artists method of user resource
         """
         user = client.get_user(359622)
+
+        # tests list
         artists = user.get_artists()
         assert isinstance(artists, list)
         artist = artists[0]
         assert isinstance(artist, deezer.resources.Artist)
         assert repr(artist) == "<Artist: Wax Tailor>"
-        assert type(user.iter_artists()) == GeneratorType
+
+        # tests generator
+        artists_generator = user.iter_artists()
+        assert type(artists_generator) == GeneratorType
+        artist = next(artists_generator)
+        assert isinstance(artist, deezer.resources.Artist)
+        assert repr(artist) == "<Artist: Wax Tailor>"
 
     def test_user_playlists(self, client):
         """
         Test playlists method of user resource
         """
         user = client.get_user(359622)
+
+        # tests list
         playlists = user.get_playlists()
         assert isinstance(playlists, list)
         playlist = playlists[0]
         assert isinstance(playlist, deezer.resources.Playlist)
         assert repr(playlist) == "<Playlist: AC/DC>"
-        assert type(user.iter_playlists()) == GeneratorType
+
+        # tests generator
+        playlists_generator = user.iter_playlists()
+        assert type(playlists_generator) == GeneratorType
+        playlist = next(playlists_generator)
+        assert isinstance(playlist, deezer.resources.Playlist)
+        assert repr(playlist) == "<Playlist: AC/DC>"
 
     def test_user_tracks(self, client):
         """
         Test tracks method of user resource
         """
         user = client.get_user(353978015)
+
+        # tests list
         tracks = user.get_tracks()
         assert isinstance(tracks, list)
         track = tracks[0]
         assert isinstance(track, deezer.resources.Track)
         assert repr(track) == "<Track: Prélude a l'après-midi d'un faune, L. 86>"
-        assert type(user.iter_tracks()) == GeneratorType
+
+        # tests generator
+        tracks_generator = user.iter_tracks()
+        assert type(tracks_generator) == GeneratorType
+        track = next(tracks_generator)
+        assert isinstance(track, deezer.resources.Track)
+        assert repr(track) == "<Track: Prélude a l'après-midi d'un faune, L. 86>"
 
 
 class TestPlaylist:
     def test_get_tracks(self, client):
+        """
+        Test tracks method of playlist resource
+        """
         playlist = client.get_playlist(12345)
+
+        # tests list
         tracks = playlist.get_tracks()
+        assert isinstance(tracks, list)
         assert len(tracks) == 4
         for track in tracks:
             assert isinstance(track, deezer.resources.Track)
         assert tracks[0].title == "Skanky Panky"
-        assert type(playlist.iter_tracks()) == GeneratorType
+
+        # tests generator
+        tracks_generator = playlist.iter_tracks()
+        assert type(tracks_generator) == GeneratorType
+        track = next(tracks_generator)
```

Comment:
Maybe the rest of this test would be more readable if we were casting it to a list before doing assertion on the content:

```
playlist_tracks = playlist.iter_tracks()
assert type(playlist_tracks) == GeneratorType
playlist_tracks = list(playlist_tracks)
assert len(playlist_tracks) == 5
assert all(isinstance(track, deezer.resources.Track) for track in playlist_tracks)
```

Reply:
LGTM - I'll edit and commit it